W-21217408 rtf and real time (#1007)
mulesoft/docs-gateway: gateway/1.12/modules/ROOT/pages/flex-architecture-managed-dr-ha.adoc

endif::[]
:imagesdir: ../assets/images

> CloudHub 2.0 and Runtime Fabric manages Managed Flex Gateway replica deployments and controls the gateway's high availability and disaster recovery. 

Reviewer: ```suggestion
CloudHub 2.0 and Runtime Fabric manage Managed Flex Gateway replica deployments and control the gateway's high availability and disaster recovery. 
```

> This documentation covers specific details of Managed Flex Gateway high availability and disaster recovery. To fully understand high availability and disaster recovery, familiarize yourself with xref:cloudhub-2::ch2-ha-dr.adoc[].

Reviewer: Avoid self-referential language.

> The following diagram shows how multiple Managed Flex Gateway replicas are deployed. External traffic entering through private space ingress is distributed across a kubernetes service of Managed Flex Gateway instances. Replicas share the gateway configuration and data that must be persistent in a Kubernetes resource. Replicas share HTTP caching and distributed rate limiting data directly with each other.

Reviewer: ```suggestion
This diagram shows how multiple Managed Flex Gateway replicas are deployed. External traffic entering through private space ingress is distributed across a kubernetes service of Managed Flex Gateway instances. Replicas share the gateway configuration and data that must persist in a Kubernetes resource. Replicas share HTTP caching and distributed rate limiting data directly with each other.
```

Reviewer: Suggest moving the sentences after the first one to below the diagram.

image:managed-high-availability.png["External traffic flows through CloudHub 2.0 Ingress to a Kubernetes service that distributes requests across multiple Managed Flex Gateway replicas"]

> NOTE: The diagram shows CloudHub 2.0 Ingress. For Runtime Fabric, you must configure your own ingress. Your ingress must terminate TLS before forwarding the request to the Managed Flex Gateway.

Reviewer: ```suggestion
NOTE: The diagram shows CloudHub 2.0 ingress. For Runtime Fabric, you must configure your own ingress. Your ingress must terminate TLS before forwarding the request to the Managed Flex Gateway.
```

== Managed Flex Gateway High Availability

The private space manages Managed Flex Gateway high availability by ensuring:

* At least two Flex replicas are always available and distributed across different availability zones in the private space. Traffic is distributed to replicas and zones evenly using a round robin traffic distribution.

** If a replica fails, a new replica is spawned. While respawning the replica, the outstanding replicas receive a temporary spike in traffic until the new replica can accept traffic. These short spikes in traffic are visible in monitoring as a temporary spike in CPU usage.
 
** Overloaded gateways don't abruptly stop incoming traffic processing as a result of overload. However, overloaded gateways show gradual degradation of response time for requests until CPU subsides.

> * Large and Extra Large Managed Flex Gateways have more capacity allocated per replica and autoscale up to three replicas across three availability zones depending on CPU usage. 

Reviewer: Is there a reason these are init cap?
